roaster-hibernate/src/main/java/hu/icellmobilsoft/roaster/hibernate/producer/EntityManagerFactoryProducer.java

```diff
@@ -0,0 +1,125 @@
+import hu.icellmobilsoft.roaster.hibernate.config.HibernateConfig;
+
+/**
+ * Producer for creating or obtaining {@link EntityManagerFactory} from myPu persistenceUnit from META-INF/persistence.xml
+ *
+ * @since 0.2.0
+ * @author speter555
+ */
+@ApplicationScoped
+public class EntityManagerFactoryProducer {
+
+    private final Logger logger = Logger.getLogger(EntityManagerFactoryProducer.class);
+
+    @Inject
+    private BeanManager beanManager;
+
+    /**
+     * Producer for creating or obtaining {@link EntityManagerFactory}
+     *
+     * @param injectionPoint
+     *            CDI injection point
+     * @return {@link EntityManagerFactory} instance
+     * 
+     * @throws BaseException
+     *             exception
+     */
+    @Produces
+    @Dependent
+    @HibernatePersistenceConfig(persistenceUnitName = "")
+    public EntityManagerFactory produceEntityManagerFactory(InjectionPoint injectionPoint) throws BaseException {
+
+        HibernatePersistenceConfig hibernatePersistenceConfig = AnnotationUtil.getAnnotation(injectionPoint, HibernatePersistenceConfig.class)
+                .orElseThrow(() -> new BaseException(CoffeeFaultType.INVALID_INPUT, "PersisteneUnitName annotation have to have configKey value!"));
+        HibernateConfig hibernateConfig = CDI.current()
+                .select(HibernateConfig.class, new HibernatePersistenceConfig.Literal(hibernatePersistenceConfig.persistenceUnitName())).get();
+
+        Map<String, Object> props = new HashMap<>();
+
+        // Set CDI Bean manager
+        props.put(Environment.CDI_BEAN_MANAGER, beanManager);
+
+        // Statistics, warning, logs
+        props.put(Environment.LOG_SESSION_METRICS, true);
+        props.put(Environment.LOG_JDBC_WARNINGS, true);
+        props.put(Environment.GENERATE_STATISTICS, true);
+
+        // JPA use in JAVA SE
+        props.put(Environment.JPA_TRANSACTION_TYPE, "RESOURCE_LOCAL");
+        props.put(Environment.JPA_PERSISTENCE_PROVIDER, "org.hibernate.jpa.HibernatePersistenceProvider");
+
+        //
+        props.put(Environment.USE_NEW_ID_GENERATOR_MAPPINGS, false);
+
+        // Set settings from Roaster config
+        props.put(Environment.DIALECT, hibernateConfig.getDialect());
+        props.put(Environment.POOL_SIZE, hibernateConfig.getPooSize());
+        props.put(Environment.SHOW_SQL, hibernateConfig.getShowSql());
+        props.put(Environment.FORMAT_SQL, hibernateConfig.getFormatSql());
+        props.put(Environment.DEFAULT_SCHEMA, hibernateConfig.getDefaultSchema());
+        props.put(Environment.JPA_JDBC_URL, hibernateConfig.getJpaJdbcUrl());
+        props.put(Environment.JPA_JDBC_USER, hibernateConfig.getJpaJdbcUser());
+        props.put(Environment.JPA_JDBC_PASSWORD, hibernateConfig.getJpaJdbcPassword());
+        props.put(Environment.JPA_JDBC_DRIVER, hibernateConfig.getJpaJdbcDriver());
+
+        return Persistence.createEntityManagerFactory(hibernatePersistenceConfig.persistenceUnitName(), props);
```

Comment:
a persistence.xml beallitasai milyen szerepet kapnak, azok felul irnak valamit, vagy aztkat irjuk felul?

Reply:
Mindenképpen szükséges az entitások megadása benne, azért szükséges csak a persistence.xml, amúgy szabadon módosítható is, viszont ebben a Producerben ezen részek mindig felül fognak íródni, mindegy hogy mi van a persistence.xml-ben.